src/components/inputs/checkbox-input/checkbox-input.js

```diff
@@ -52,40 +63,38 @@ class CheckboxInput extends React.PureComponent {
 
   render() {
     return (
-      <div>
-        <label htmlFor={this.state.id} css={getLabelStyles(this.props)}>
-          <Spacings.Inline alignItems="center">
-            <div css={getCheckboxWrapperStyles(this.props)}>
-              {(() => {
-                if (this.props.isIndeterminate) return <Icons.Indeterminate />;
-                if (this.props.isChecked) return <Icons.Checked />;
-                return <Icons.Unchecked />;
-              })()}
-            </div>
-            {this.props.children && (
-              <Text.Body
-                // FIXME: add proper tones when we have disabled/primary in tones
-                tone={this.props.isDisabled ? 'secondary' : undefined}
-              >
-                {this.props.children}
-              </Text.Body>
-            )}
-            <input
-              css={css`
-                display: none;
-              `}
-              type="checkbox"
-              id={this.state.id}
-              name={this.props.name}
-              value={this.props.value}
-              onChange={this.props.onChange}
-              disabled={this.props.isDisabled}
-              checked={this.props.isChecked && !this.props.isIndeterminate}
-              {...filterDataAttributes(this.props)}
-            />
-          </Spacings.Inline>
-        </label>
-      </div>
+      <Label htmlFor={this.state.id} hasError={this.props.hasError}>
+        <Spacings.Inline alignItems="center">
+          <Checkbox
+            type="checkbox"
+            id={this.state.id}
+            name={this.props.name}
+            value={this.props.value}
+            onChange={this.props.onChange}
+            disabled={this.props.isDisabled}
+            checked={this.props.isChecked && !this.props.isIndeterminate}
```

Comment:
Why do we remap the props here already to "native" DOM input props? Can't we pass them to our checkbox, maintaining our naming convention and then explicitely pass them on to the input?

Reply:
yeah sure 👍